dashboard/modules/reporter/profile_manager.py

```diff
@@ -49,33 +49,46 @@ def _format_failed_pyspy_command(cmd, stdout, stderr) -> str:
 # If we can sudo, always try that. Otherwise, py-spy will only work if the user has
 # root privileges or has configured setuid on the py-spy script.
 async def _can_passwordless_sudo() -> bool:
-    process = await asyncio.create_subprocess_shell(
-        "sudo -n true",
+    process = await asyncio.create_subprocess_exec(
+        "sudo", "-n", "true",
         stdout=subprocess.PIPE,
         stderr=subprocess.PIPE,
-        shell=True,
     )
     _, _ = await process.communicate()
     return process.returncode == 0
 
 
+async def _get_pyspy_path() -> str:
+    process = await asyncio.create_subprocess_exec(
+        "which", "py-spy",
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+    stdout, _ = await process.communicate()
+    if process.returncode != 0:
+        return ""
+    return stdout.decode().strip()
```

Comment:
Can this be? https://docs.python.org/3/library/shutil.html#shutil.which

Reply:
Done! Great lib!